File: CS/DiagnosticTool/Data/SessionItem.cs
using System.Xml.Serialization;

namespace DashboardDiagnosticTool.Data {
    public class SessionItem : ItemBase {
        [XmlAttribute]
        public int ID => SessionId;
        public int ParentID => SessionId;

        [XmlAttribute]
        public string Name { get; set; }
        public SessionItem() { }
        public SessionItem(int sessionId) : base(sessionId, -1) {
            Name = $"Session{sessionId}";
        }

        public override int GetHashCode() {
            return ID.GetHashCode();
        }
    }
}

File: VB/DiagnosticTool/Data/SessionItem.vb
Imports System.Xml.Serialization

Namespace DashboardDiagnosticTool.Data

    Public Class SessionItem
        Inherits ItemBase

        <XmlAttribute>
        Public ReadOnly Property ID As Integer
            Get
                Return SessionId
            End Get
        End Property

        Public ReadOnly Property ParentID As Integer
            Get
                Return SessionId
            End Get
        End Property

        <XmlAttribute>
        Public Property Name As String

        Public Sub New()
        End Sub

        Public Sub New(ByVal sessionId As Integer)
            MyBase.New(sessionId, -1)
            Name = $"Session{sessionId}"
        End Sub

        Public Overrides Function GetHashCode() As Integer
            Return ID.GetHashCode()
        End Function
    End Class
End Namespace
